### PR TITLE
Enable esModuleInterop Repo Wide 

### LINKED ISSUE #6126: Enable esModuleInterop Repo Wide
Some packages like `isutf8` require the flag to work nicely, and it's the default of Babel, including the default Metro preset. We should look at turning it on everywhere, but need to be careful since this could potentially cause runtime breaks.

### PR DESCRIPTION
**Pull request by NickGerleman**
Fixes #6126

esModuleInterop changes the interaction model on how ES6 import in TypeScript is translated into the CommonJS equivalent. It more specifically allows for full modules to be imported via default import (via implictly enabled `allowSyntheticDefaultImports`), This is the default of Babel, but not `tsc`, which leads to inconsistent support in debug vs release, or with some TS code in the community.

I've changed this flag, and CommonJS imports, which also removes the ability to namespace import exported CommonJS functions (requiring default import instead),

We similarly turn on `isolatedModules` to ensure babel compat.

I fixed any TS build errors, but there are some cases where `tsc` is unaware of runtime errors. E.g. where a module we import is declared as an esModule, but does not have a generated default export on it (such as TS without esModuleInterop).  We may have some regression in the case of incorrect typings combined with lack of runtime tests. Stock CommonJS modules, and TS compiled with esModuleInterop should work correctly though.

I tried to unify tsconfig as well, removing some vestigies like compilng TS platforms overrides as ES5.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/7900)

### CODE REVIEW

File: vnext/tsconfig.json [hunk 1]
```diff
@@ -1,19 +1,8 @@
 {
+  "extends": "@rnw-scripts/ts-config",
   "compilerOptions": {
     "baseUrl": ".",
-    "target": "es6",
```
Comment (NickGerleman) [inline vnext/tsconfig.json:4]: This was overriden to ES5 earlier, but new target is ES2018, which I think Metro with new Babel should be able to handle fine.

**Review by dannyvv — APPROVED** [review]
Looks like we already had
```json
  "engines": {
    "node": ">12.0.0"
  }
```
 in our `/package.json` we had >= 12.0.0 in pacages\react-native-platform-override

Not needed for this PR, but should we consider uniyfing the expression to `>=` and also add it to our CLI?

**Comment by NickGerleman** [comment]
@msftbot merge in 10 hours

**Comment by NickGerleman** [comment]
> Looks like we already had
> 
> ```json
>   "engines": {
>     "node": ">12.0.0"
>   }
> ```
> 
> in our `/package.json` we had >= 12.0.0 in pacages\react-native-platform-override
> 
> Not needed for this PR, but should we consider uniyfing the expression to `>=` and also add it to our CLI?

Will create an issue

**Comment by NickGerleman** [comment]
https://github.com/microsoft/react-native-windows/issues/7912

**Pull request merged**